packages/dev/docs/src/Layout.js

```diff
@@ -115,37 +129,105 @@ export function Layout({scripts, styles, pages, currentPage, publicUrl, children
+    let pageDir = pageParts[0];
+
+    // Pages within same directory (react-spectrum/Alert.html)
+    if (currentParts.length > 1) {
+      return currentDir === pageDir && !isIndex.test(p.name);
+    }
+
+    // Top-level index pages (react-spectrum/index.html)
+    if (currentParts.length === 1 && pageParts.length > 1 && isIndex.test(p.name)) {
+      return true;
+    }
+
+    // Other top-level pages
+    return !isIndex.test(p.name) && pageParts.length === 1;
+  });
+
+  let title = currentParts.length > 1 ? dirToTitle(currentPageName) : 'React Spectrum';
+
+  return (
+    <nav className={docStyles.nav}>
+      <header>
+        {currentParts.length > 1 &&
+          <a href="../index.html" className={docStyles.backBtn}>
+            <ChevronLeft />
+          </a>
+        }
+        <a href={publicUrl} className={docStyles.homeBtn}>
+          <svg viewBox="0 0 30 26" fill="#E1251B">
+            <polygon points="19,0 30,0 30,26" />
+            <polygon points="11.1,0 0,0 0,26" />
+            <polygon points="15,9.6 22.1,26 17.5,26 15.4,20.8 10.2,20.8" />
+          </svg>
+          <h2 className={typographyStyles['spectrum-Heading4']}>
+            {title}
+          </h2>
+        </a>
+      </header>
+      <ul className={sideNavStyles['spectrum-SideNav']}>
+        {pages.map(p => (
+          <li className={classNames(sideNavStyles['spectrum-SideNav-item'], {[sideNavStyles['is-selected']]: p.name === currentPageName})}>
+            <a className={sideNavStyles['spectrum-SideNav-itemLink']} href={p.url}>{p.title}</a>
+          </li>
+        ))}
+      </ul>
+    </nav>
+  );
+}
+
+function Footer() {
+  const year = new Date().getFullYear();
+  return (
+    <footer>
+      <hr className={classNames(ruleStyles['spectrum-Rule'], ruleStyles['spectrum-Rule--small'], ruleStyles['spectrum-Rule--horizontal'])} />
+      <ul>
+        <li>Copyright © {year} Adobe. All rights reserved.</li>
+        <li><a className={linkStyle['spectrum-Link--secondary']} href="//www.adobe.com/privacy.html">Privacy</a></li>
+        <li><a className={linkStyle['spectrum-Link--secondary']} href="//www.adobe.com/legal/terms.html">Terms of Use</a></li>
+        <li><a className={linkStyle['spectrum-Link--secondary']} href="//www.adobe.com/privacy/cookies.html">Cookies</a></li>
+      </ul>
+    </footer>
+  );
+}
+
+export function Layout({scripts, styles, pages, currentPage, publicUrl, children, toc}) {
+
+  return (
+    <Page title={currentPage.title} scripts={scripts} styles={styles}>
+      <div className={docStyles.pageHeader} id="header" />
+      <Nav currentPageName={currentPage.name} pages={pages} publicUrl={publicUrl} />
+      <main>
+        <article className={typographyStyles['spectrum-Typography']}>
+          <MDXProvider components={mdxComponents}>
+            <LinkProvider>{children}</LinkProvider>
+          </MDXProvider>
+        </article>
+        {toc.length ? <ToC toc={toc} /> : null}
+        <Footer />
```

Comment:
the styling works best this way, but im not sure if a footer should reside inside the `main` element? if we want it out, we will likely have to convert it to a div that has the current styling of the main element, so that the footer lines up correctly.
And then have the main element just be a semantic container with no styles.